Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp

```diff
@@ -0,0 +1,205 @@
+  using ValueType = value_type;
+
+  Adapter(SpacePointCollection&&) = delete;
+  Adapter(const SpacePointCollection& externalCollection)
+      : m_storage(&externalCollection) {}
+
+ private:
+  std::size_t size_impl() const { return storage().size(); }
+
+  float x_impl(std::size_t idx) const { return storage()[idx].x; };
+  float y_impl(std::size_t idx) const { return storage()[idx].y; };
+  float z_impl(std::size_t idx) const { return storage()[idx].z; };
+  float varianceR_impl(std::size_t idx) const { return storage()[idx].varR; }
+  float varianceZ_impl(std::size_t idx) const { return storage()[idx].varZ; }
+
+  const SpacePoint& get_impl(std::size_t idx) const { return storage()[idx]; }
+
+  std::any component_impl(Acts::HashedString key, std::size_t /*n*/) const {
+    using namespace Acts::HashedStringLiteral;
+    switch (key) {
+      case "TopStripVector"_hash:
+      case "BottomStripVector"_hash:
+      case "StripCenterDistance"_hash:
+      case "TopStripCenterPosition"_hash:
+        return Acts::Vector3(0, 0, 0);
+      default:
+        throw std::runtime_error("no such component " + std::to_string(key));
+    }
+  }
+
+  const SpacePointCollection& storage() const { return *m_storage; }
+
+ private:
+  const SpacePointCollection* m_storage{};
+};
+
+BOOST_AUTO_TEST_CASE(spacepoint_container_edm_traits) {
+  using adapter_t = Acts::Test::Adapter;
+  using container_t =
+      Acts::SpacePointContainer<adapter_t, Acts::detail::RefHolder>;
+  using proxy_t = Acts::SpacePointProxy<container_t>;
+  using iterator_t = Acts::ContainerIndexIterator<container_t, proxy_t&, false>;
+
+  static_assert(std::ranges::range<container_t>);
+  static_assert(std::same_as<typename iterator_t::iterator_category,
+                             std::random_access_iterator_tag>);
+  static_assert(
+      std::same_as<typename std::iterator_traits<iterator_t>::iterator_category,
+                   std::random_access_iterator_tag>);
+}
+
+BOOST_AUTO_TEST_CASE(spacepoint_container_edm_constructors) {
+  std::size_t nExternalPoints = 10;
+  SpacePointCollection externalCollection(nExternalPoints);
+
+  Acts::SpacePointContainerConfig spConfig;
+  Acts::SpacePointContainerOptions spOptions;
+
+  Acts::Test::Adapter adapterForRef(externalCollection);
+  Acts::SpacePointContainer<Acts::Test::Adapter, Acts::detail::RefHolder>
+      spContainerRef(spConfig, spOptions, adapterForRef);
+
+  Acts::SpacePointContainer<Acts::Test::Adapter, Acts::detail::ValueHolder>
+      spContainerVal(spConfig, spOptions,
+                     Acts::Test::Adapter(externalCollection));
+
+  BOOST_CHECK_EQUAL(spContainerRef.size(), nExternalPoints);
+  BOOST_CHECK_EQUAL(spContainerVal.size(), nExternalPoints);
+}
+
+BOOST_AUTO_TEST_CASE(spacepoint_container_edm_functionalities) {
+  std::size_t nExternalPoints = 100;
+  SpacePointCollection externalCollection;
+  externalCollection.reserve(nExternalPoints);
+  for (std::size_t i(0); i < nExternalPoints; ++i) {
```

Comment:
```suggestion
  for (std::size_t i = 0; i < nExternalPoints; ++i) {
```
This is more common in ACTS and we try to get away from the `()` initialisation.

Reply:
done